packager/mpd/base/xml/xml_node.cc

```diff
@@ -53,6 +54,22 @@ const char kDTSCCodec[] = "dtsc";
 const char kDTSECodec[] = "dtse";
 const char kDTSXCodec[] = "dtsx";
 
+std::string urlEncode(const std::string& input) {
+  CURL* curl = curl_easy_init();  // Initialize a CURL session
+  if (curl) {
+    char* output = curl_easy_escape(curl, input.c_str(), input.length());
```

Comment:
According to the `curl_easy_escape` docs:

> Since 7.82.0, the curl parameter is ignored. Prior to that there was per-handle character conversion support for some old operating systems such as TPF, but it was otherwise ignored.

So I think it's best if we don't allocate and free the `CURL*` handle, just to cut down on memory churn and risk of leaking if an exception is thrown.

Reply:
Done.  PTAL